aspnetcore/fundamentals/minimal-apis/7.0-samples/MinApiTests/UnitTests/TodoInMemoryTests.cs

```diff
@@ -0,0 +1,135 @@
+using Microsoft.AspNetCore.Http.HttpResults;
+using todo_group;
+using todo_group.Data;
+
+namespace MinApiTests.UnitTests
+{
+    public class TodoInMemoryTests
+    {
+        [Fact]
+        public async Task GetTodoReturnsTodoFromDatabase()
+        {
+            // Arrange
+            await using var context = new MockDb().CreateDbContext();
+
+            context.Todos.Add(new Todo
+            {
+                Id = 1,
+                Title = "Test title",
+                Description = "Test description",
+                IsDone = false
+            });
+
+            await context.SaveChangesAsync();
+
+            // Act
+            var okResult = (Ok<Todo>)await TodoEndpointsV1.GetTodo(1, context);
+            var notFoundResult = (NotFound)await TodoEndpointsV1.GetTodo(404, context);
+
+            //Assert
+            Assert.Equal(200, okResult.StatusCode);
+            var foundTodo = Assert.IsAssignableFrom<Todo>(okResult.Value);
+            Assert.Equal(1, foundTodo.Id);
+
+            Assert.Equal(404, notFoundResult.StatusCode);
+        }
+
+        [Fact]
+        public async Task CreateTodoCreatesTodoInDatabase()
+        {
+            //Arrange
+            await using var context = new MockDb().CreateDbContext();
+
+            var newTodo = new TodoDto
+            {
+                Title = "Test title",
+                Description = "Test description",
+                IsDone = false
+            };
+
+            //Act
+            var createdResult = (Created<Todo>)await TodoEndpointsV1.CreateTodo(newTodo, context);
+
+            //Assert
+            Assert.Equal(201, createdResult.StatusCode);
+            Assert.NotNull(createdResult.Location);
+            Assert.IsAssignableFrom<Todo>(createdResult.Value);
+
+            Assert.NotEmpty(context.Todos);
+            Assert.Collection(context.Todos, todo =>
+            {
+                Assert.Equal("Test title", todo.Title);
+                Assert.Equal("Test description", todo.Description);
+                Assert.False(todo.IsDone);
+            });
+        }
+
+        [Fact]
+        public async Task UpdateTodoUpdatesTodoInDatabase()
+        {
+            //Arrange
+            await using var context = new MockDb().CreateDbContext();
+
+            context.Todos.Add(new Todo
+            {
+                Id = 1,
+                Title = "Exiting test title",
+                IsDone = false
+            });
+
+            await context.SaveChangesAsync();
+
+            var updatedTodo = new Todo
+            {
+                Id = 1,
+                Title = "Updated test title",
+                IsDone = true
+            };
+
+            //Act
+            var createdResult = (Created<Todo>)await TodoEndpointsV1.UpdateTodo(updatedTodo, context);
+            var notFoundResult = (NotFound)await TodoEndpointsV1.UpdateTodo(new Todo { Id = 2, Title = "Invalid Title" }, context);
```

Comment:
What do you think about splitting the tests?

Reply:
I think for documented examples and as a unit, this is ok. If I separate these out there will be three more test methods.

Reply:
3 tests would be better.

Reply:
I've added another fact just to check whether the todo is available inside the db,

`TodoInMemoryTests.cs` (When DbContext is injected)

```
[Fact]
public async Task GetTodoReturnsNotFoundIfNotExists()
{
    // Arrange
    await using var context = new MockDb().CreateDbContext();

    // Act
    var notFoundResult = (NotFound)await TodoEndpointsV1.GetTodo(404, context);

    //Assert
    Assert.Equal(404, notFoundResult.StatusCode);
}
```

`TodoMoqTests.cs` (When a service wrapper of the DbContext is injected)
```
[Fact]
public async Task GetTodoReturnsNotFoundIfNotExists()
{
    // Arrange
    var mock = new Mock<ITodoService>(); 
    
    mock.Setup(m => m.Find(It.Is<int>(id => id == 404)))
        .ReturnsAsync((Todo?)null);

    // Act
    var notFoundResult = (NotFound)await TodoEndpointsV2.GetTodo(404, mock.Object);

    //Assert
    Assert.Equal(404, notFoundResult.StatusCode);
}
```